cloudflare/src/framework/async_api.rs

```diff
@@ -21,6 +24,16 @@ pub trait ApiClient {
         ResultType: ApiResult,
         QueryType: Serialize,
         BodyType: Serialize;
+
+    /// Send a request to a particular Cloudflare API endpoint, get the response as bytes.
+    async fn request_raw_bytes<ResultType, QueryType, BodyType>(
```

Comment:
I don't agree with this notion that "any endpoint can be requested to return its actual return type, or to return a bunch of bytes". This feels like sloppy design to me. The reality of our API is that an endpoint always knows whether it returns a bunch of bytes or something else. That fact must be reflected in our API client design, there's no point in giving the user a choice between two functions to invoke.

Reply:
I agree that some endpoints should never be requested as JSON. But I think it's reasonable for JSON endpoints to be requested as bytes. Some users may use cloudflare-rs for the strong type checking of endpoints and parameters, but want byte output. Or maybe they want to apply some function between receiving the response and deserializing the JSON (e.g. logging, filtering).